src/airtable.ts

```diff
@@ -0,0 +1,92 @@
+import Base from './base';
+import Record from './record';
+import Table from './table';
+import AirtableError from './airtable_error';
+
+class Airtable {
+    readonly _apiKey: string;
+    readonly _endpointUrl: string;
+    readonly _apiVersion: string;
+    readonly _apiVersionMajor: string;
+    readonly _noRetryIfRateLimited: boolean;
```

Comment:
You can argue otherwise but I consider that a backwards incompatible change. I suppose it might be a minor breaking change instead of major breaking change. But I want to avoid any mechanical change in this changeset.

-----

Beyond this PR, I agree, switching to private fields would be good here.

Reply:
Marking these as `private` creates typescript errors in `base.ts`.

> Property '_endpointUrl' is private and only accessible within class 'Airtable'. ts(2341)

So we can't mark these properties in `airtable.ts` as `private` without changing the executed relationship in `base.ts`. That probably means some kind of API refactor?